tempesta_fw/sock_srv.c

```diff
@@ -174,19 +184,15 @@ tfw_sock_srv_connect_try(TfwSrvConn *srv_conn)
 			TFW_ERR("Unable to initiate a connect to server: %d\n",
 				r);
 		ss_close_sync(sk, false);
+		/* Drop failed connection. */
+		srv_conn->proto.hooks->connection_drop(sk);
```

Comment:
It appears that this is not very clean. For instance, where's a call to `tfw_connection_unlink_from_sk()`? Also, calling `connection_drop()` is not needed when a connection is not live -- see `tfw_sock_srv_connect_failover()`. Also, doing it this way unnecessarily increments the stats count of disconnects.